appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java

```diff
@@ -356,66 +356,86 @@ public void visitConsumes(Consumes consumes, AnnotatedElement element, ApiContex
 
-        // Set the request body type accordingly.
-        context.getWorkingOperation().getRequestBody().getContent().get(javax.ws.rs.core.MediaType.WILDCARD).getSchema()
-                .setType(formSchemaType);
+            // Set the request body type accordingly.
+            context.getWorkingOperation().getRequestBody().getContent().get(javax.ws.rs.core.MediaType.WILDCARD).getSchema()
+                    .setType(formSchemaType);
+        }
     }
 
     @Override
-    public void visitHeaderParam(HeaderParam param, java.lang.reflect.Parameter element, ApiContext context) {
+    public void visitHeaderParam(HeaderParam param, AnnotatedElement element, ApiContext context) {
         org.eclipse.microprofile.openapi.models.parameters.Parameter newParameter = new ParameterImpl();
         newParameter.setName(param.value());
         newParameter.setIn(In.HEADER);
-        newParameter.setStyle(Style.SIMPLE);
-        newParameter.setSchema(new SchemaImpl().type(ModelUtils.getSchemaType(element.getType())));
-        context.getWorkingOperation().addParameter(newParameter);
+        addParameter(element, context, newParameter);
     }
 
     @Override
-    public void visitCookieParam(CookieParam param, java.lang.reflect.Parameter element, ApiContext context) {
+    public void visitCookieParam(CookieParam param, AnnotatedElement element, ApiContext context) {
         org.eclipse.microprofile.openapi.models.parameters.Parameter newParameter = new ParameterImpl();
         newParameter.setName(param.value());
         newParameter.setIn(In.COOKIE);
-        newParameter.setStyle(Style.SIMPLE);
-        newParameter.setSchema(new SchemaImpl().type(ModelUtils.getSchemaType(element.getType())));
-        context.getWorkingOperation().addParameter(newParameter);
+        addParameter(element, context, newParameter);
+    }
+    
+      private void addParameter(AnnotatedElement element, ApiContext context, org.eclipse.microprofile.openapi.models.parameters.Parameter newParameter) {
+        if (element instanceof java.lang.reflect.Parameter) {
+            newParameter.setSchema(new SchemaImpl().type(ModelUtils
+                    .getSchemaType(java.lang.reflect.Parameter.class.cast(element).getType())));
+        } else {
+            newParameter.setSchema(new SchemaImpl().type(ModelUtils
+                    .getSchemaType(Field.class.cast(element).getType())));
+        }
+        
+        if (context.getWorkingOperation() != null) {
+            context.getWorkingOperation().addParameter(newParameter);
+        } else {
+            Field field = Field.class.cast(element);
+            for (Method method : field.getDeclaringClass().getDeclaredMethods()) {
+                OpenAPI api = context.getApi();
+                ApiContext apiContext = new OpenApiContext(api, null, ModelUtils.getOperation(method, api, generateResourceMapping(classes)));
+                apiContext.getWorkingOperation().addParameter(newParameter);
```

Comment:
I'm confused as to how this works, isn't this context just thrown away once it's used?